base/env.jl

```diff
@@ -7,8 +7,10 @@ if Sys.iswindows()
     _hasenv(s::Vector{UInt16}) = _getenvlen(s) != 0 || Libc.GetLastError() != ERROR_ENVVAR_NOT_FOUND
     _hasenv(s::AbstractString) = _hasenv(cwstring(s))
 
+    const JULIA_DEBUG_CWSTRING = cwstring("JULIA_DEBUG")
+
     function access_env(onError::Function, str::AbstractString)
-        var = cwstring(str)
+        var = str == "JULIA_DEBUG" ? JULIA_DEBUG_CWSTRING : cwstring(str)
         len = _getenvlen(var)
         if len == 0
```

Comment:
I think we could do this conversion at callsite? Having a string compare here seems less than ideal.

Reply:
Or we could teach the compiler to constprop through cwstring.

Reply:
Since `cwstring` is not consistent, that may be fairly difficult. This SGTM.

Reply:
Alternatively, we could replace all our code with the libuv wrapper for the same, which does a bit on the stack before switching to malloc for the string conversion.